### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by yasahi-hpc**
This PR aims at final updating of docs before release.

1. files under `docs` updated based on recent changes
2. README.md updated
3. fix issues in Doxygen

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -111,11 +111,10 @@ cmake -DCMAKE_CXX_COMPILER=g++ \
 This way, all the functionalities are executed on A100 GPUs.
```
Comment (pzehner) [inline README.md:111]: I think it's not the best idea to duplicate install instructions on both the readme and on the doc. This is an error-prone process (DRY). Since GitHub can display RST files, a different approach is possible.

You should have a condensed section on how to use KokkosFFT as a subdirectory dependency (this is likely how a programmer would start using this library), then have an anchor to the documentation where the complete install process is given:

```md
Further installation details are provided in the [documentation](./docs/intro/building.rst).
```

I think this approach could be used for other sections of the readme. In other words, this file (which is the first contact for a potential user) would be a simple starter, redirecting to more complete documentation pages.

File: docs/intro/building.rst [hunk 2]
```diff
@@ -63,7 +63,8 @@ The code can be built as
 .. code-block:: bash
 
     mkdir build && cd build
-    cmake -DCMAKE_PREFIX_PATH="<install_dir>/kokkos;<install_dir>/kokkos_fft" ..
+    cmake -DCMAKE_CXX_COMPILER=icpx \
```
Comment (pzehner) [inline docs/intro/building.rst:66]: Why advertise for the Intel compiler?

You could be more generic with `-DCMAKE_CXX_COMPILER=<your C++ compiler>`.

File: docs/intro/building.rst [hunk 3]
```diff
@@ -22,7 +22,7 @@ However, we have not tested all the listed compilers there and thus recommend th
 Install KokkosFFT as a library
 ------------------------------
 
-Is is assumed that Kokkos is installed under ``<install_dir>/kokkos`` with OpenMP backend. Here is a recipe to install KokkosFFT under ``<install_dir>/kokkos_fft``.
+Is is assumed that Kokkos is installed under ``<install_dir>/kokkos`` with OpenMP backend. Here is a recipe to install KokkosFFT under ``<install_dir>/kokkosFFT``.
```
Comment (pzehner) [inline docs/intro/building.rst:25]: I think you can rephrase this as:

```rst
Let's assume Kokkos is installed under ``path/to/kokkos`` with OpenMP backend. We build and install KokkosFFT under ``path/to/kokkos-fft``.
```

File: docs/intro/building.rst [hunk 4]
```diff
@@ -63,7 +63,8 @@ The code can be built as
 .. code-block:: bash
 
     mkdir build && cd build
-    cmake -DCMAKE_PREFIX_PATH="<install_dir>/kokkos;<install_dir>/kokkos_fft" ..
+    cmake -DCMAKE_CXX_COMPILER=icpx \
+          -DCMAKE_PREFIX_PATH="<install_dir>/kokkos;<install_dir>/kokkosFFT" ..
```
Comment (pzehner) [inline docs/intro/building.rst:67]: This line would become

```
    -DKMAKE_PREFIX_PATH=path/to/kokkos;path/to/kokkos-fft`
```

Alternatively, you can use the syntax:

```
    -DKokkos_ROOT=path/to/kokkos
    -DKokkosFFT_ROOT=path/to/kokkos-fft
```

The latest version of the cheat sheet tends to use this way.

**Review by pzehner — CHANGES_REQUESTED** [review]
This is a nice update, but I think that information in the documentation shouldn't be duplicated in the readme.

File: docs/intro/building.rst (hunk 2 above)
Comment (yasahi-hpc) [reply docs/intro/building.rst:66]: Then `g++`? I would like to make it as easy as possible, i.e., ideally, copy and paste this part is sufficient.

File: docs/intro/building.rst (hunk 3 above)
Comment (yasahi-hpc) [reply docs/intro/building.rst:25]: Sure

File: docs/intro/building.rst (hunk 4 above)
Comment (yasahi-hpc) [reply docs/intro/building.rst:67]: In the future release, we may use 

```
    -DKokkos_ROOT=path/to/kokkos
    -DKokkosFFT_ROOT=path/to/kokkos-fft
```


For the moment, we use the `-DKMAKE_PREFIX_PATH` version in CI, so I would keep in the docs as well

File: README.md (hunk 1 above)
Comment (yasahi-hpc) [reply README.md:111]: It is indeed a duplicate if people read both `README.md` and docs. However, people may not necessarily read the docs. Not sure what is the right way.

File: docs/intro/building.rst (hunk 2 above)
Comment (pzehner) [reply docs/intro/building.rst]: Then don't put anything. CMake would check the environment variable `CC`, and ultimately default to GCC.

File: docs/intro/building.rst (hunk 4 above)
Comment (pzehner) [reply docs/intro/building.rst]: Ok.

File: docs/intro/building.rst (hunk 2 above)
Comment (yasahi-hpc) [reply docs/intro/building.rst]: I fixed accordingly. I will merge this and make a release candidate. Thank you for your review

**Comment by yasahi-hpc** [comment]
Close this PR, because of asyucrhouns commits issue.
